sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/FileDataSourceV2.scala

```diff
@@ -59,4 +62,27 @@ trait FileDataSourceV2 extends TableProvider with DataSourceRegister {
     val fs = hdfsPath.getFileSystem(sparkSession.sessionState.newHadoopConf())
     hdfsPath.makeQualified(fs.getUri, fs.getWorkingDirectory).toString
   }
+
+  override def fromOptions(options: CaseInsensitiveStringMap): Identifier = {
+    val paths = getPaths(options)
+    if (paths.length > 1) {
+      throw new IllegalArgumentException("Cannot create a table in multiple locations")
+    } else if (paths.isEmpty) {
+      throw new IllegalArgumentException("Didn't specify the 'path' for file based table")
+    }
+    Identifier.of(Array.empty, paths.head)
```

Comment:
This should be a different class, `PathIdentifier`, so that we can easily identify these and handle them separately.